playbooks/web.yml

```diff
@@ -0,0 +1,120 @@
+---
+
+- hosts: beaker-project.osci.io
+  vars:
+    device: /dev/vdb
+  tasks:
+    - name: "Add lvm on data disk"
+      filesystem:
+        dev: "{{ device }}"
+        fstype: lvm
+
+    # use LVM in case we need to add more storage and divide it
+    - name: "Create a volume group on data disk"
+      lvg:
+        pvs: "{{ device }}"
+        vg: "{{ data_vg_name }}"
+
+  tags: partitioning
+
+
+- hosts: beaker-project.osci.io
+  tasks:
+    - name: "Install NFS tools"
+      package:
+        name: nfs-utils
+        state: present
+
+    - name: "Mount NFS storage"
+      mount:
+        src: "lucille.srv.osci.io:/{{ inventory_hostname }}"
+        path: /srv/data
+        fstype: nfs4
+        opts: "defaults,nodev,nosuid,noatime,nodiratime,acl,x-systemd.automount"
+        state: mounted
+
+  tags: nfs
+
+
+- hosts: beaker-project.osci.io
+  vars:
+    website_domain: www.beaker-project.org
+    document_root: "/var/www/{{ website_domain }}"
+    uploader_login: beaker-web-uploader
+    uploader_key: "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAtivtuKUWeOg5DzMtUBe6JuPzMRDhOpvhOWydXiuXxJ1tKWQoq23CxRWcelTKnsGFAHcDWvl7MGKFDQaK29MnlbF6PEXM+Rx+qSsAPSZtVHecq88zrlgoI0eDDfS7w75QkxHcFffUBNCX2d7ONTboxzFYNhVwTr08ZRRO9llvCekLKMFQha0Tco2EKSNDhyyuCsGu414JjLLdtD+JDcBSrITkz5/N7tzShq94iaVJJF5L1yV/FppT14cJIDbDX55fXB5QG/GpcXsnVt6BEotzAzrvksTVlAmmm58ZOX9zFhLEMeQh5HL8chUQS418AYy5Lq8o/n0rffYpUn5OszysRQ== jenkins@beaker-jenkins.rhev-ci-vms.eng.rdu2.redhat.com"
+    data_dir: "{{ inventory_dir }}/data/tenants/po4a/web"
+  tasks:
+    - name: "Create web storage"
+      include_role:
+        name: lvm_partition
+      vars:
+        vg: "{{ data_vg_name }}"
+        size: 100%PVS
+        lv_name: www
+        path: "/var/www/{{ website_domain }}"
+      tags: partitioning
+
+    - name: "Create uploader's user"
+      user:
+        name: "{{ uploader_login }}"
+        comment: "Website Uploader User"
+
+    - name: "Install rsync for the website builder"
+      package:
+        name: rsync
+        state: installed
+
+    - name: "Install sync SSH key"
+      authorized_key:
+        key: "{{ uploader_key }}"
+        key_options: "command=\"rsync --server -vlogtrze.isf --omit-dir-times --delete --delete-after . /var/www/{{ website_domain }}\",no-port-forwarding,no-agent-forwarding,no-X11-forwarding,no-pty"
```

Comment:
-e, --rsh=COMMAND           specify the remote shell to use
Do we need this one?

Reply:
Ok, after further investigation. This will not work for us. We are defining different attributes in different Jenkins jobs and it can't be generalized.